src/auto-evo/steps/FindBestMigration.cs

```diff
@@ -36,6 +36,8 @@ protected override IAttemptResult TryCurrentVariant()
         {
             var config = new SimulationConfiguration(map, Constants.AUTO_EVO_VARIANT_SIMULATION_STEPS);
 
+            config.SetPatchesToRunBySpeciesPresence(species);
+
             PopulationSimulation.Simulate(config);
```

Comment:
Is it safe to say that you should ALWAYS call SetPatchesToRunBySpeciesPresence() prior to running PopulationSimulation.Simulate(config)?

If so, the latter should call the former internally, to prevent misuse.

Reply:
You don't have to. For example when simulating the final populations all species are simulated at once for all patches.

Reply:
Maybe a parameter on the Simulate method then? This is something that would be very easy to miss if a future developer copied off of the time you don't need to use the config in this way as a template for a case where you do have to do this.

Reply:
They should read the simulation config, this is explained in there.